bindings/matrix-sdk-ffi/src/room_directory_search.rs

```diff
@@ -87,9 +88,15 @@ impl RoomDirectorySearch {
         Ok(())
     }
 
-    pub async fn search(&self, filter: Option<String>, batch_size: u32) -> Result<(), ClientError> {
+    pub async fn search(
+        &self,
+        filter: Option<String>,
+        batch_size: u32,
+        server: Option<String>,
```

Comment:
Random thought during a quick skim: As this parameter isn't strongly typed in the FFI, it might be worthwhile calling this `server_name` (or perhaps `remote_server_name`) to indicate that you're not meant to pass a homeserver URL here.

Reply:
Good call, done in https://github.com/matrix-org/matrix-rust-sdk/pull/4188/commits/7cb15aa2423956c8bd7f2ed21e563bb24f164535.